ide/src/main/java/com/devonfw/tools/ide/url/updater/docker/DockerDesktopUrlUpdater.java

```diff
@@ -19,37 +24,27 @@ protected String getTool() {
   @Override
   protected void addVersion(UrlVersion urlVersion) {
 
+    VersionIdentifier vid = VersionIdentifier.of(urlVersion.getName());
     String version = urlVersion.getName().replaceAll("\\.", "");
     // get Code for version
     String body = doGetResponseBodyAsString("https://docs.docker.com/desktop/release-notes/");
-    String regex = "<h2 id=\"" + version + "\".*?https://desktop\\.docker\\.com.*?(\\d{5})\\.exe(?!.<h2 id=)";
+                  // 1......2.........3...................4.....5.....................................6..............7...
+    String regex = "href=#"+version+".{8,12}(\r\n|\r|\n).{0,350}href=https://desktop\\.docker\\.com.*?(\\d{5,6}).*\\.exe";
```

Comment:
regular expression groups are indicated by parenthesis so in other words surrounded by `(` and `)`.
Therefore your group numbers are wrong and misleading here.
The `1` has to go to the beginning of `(\r\n|\r|\n)` and the 2 has to go to the beginning of `(\\d{5,6})`.
That explains why you are using group `2` for the code. There are no groups `3-7`!

BTW: We have rewritten various URL updaters to go away from HTML crawling and parsing.
This again shows how fragile this approach is.